source/particle/world.cc

```diff
@@ -165,7 +112,7 @@ namespace aspect
                   ExcMessage("A part of the model has requested access to the ghost "
                              "particles, but the parameter 'Update ghost particles' has not "
                              "been set, therefore no ghost particles are available."));
-      return ghost_particles;
+      return particle_handler->get_ghost_particles();
```

Comment:
we should eventually convert the places that use these ghost particles to call something like `particle_handler.begin_ghost()` and `particle_handler.end_ghost()`, and then use iterators as well. This would allow you to hide the underlying data structure the same way you do for the locally owned particles. 

Not sure whether that's easy to do at the moment, though.